pkg/model/utxo/milestone_diff.go

```diff
@@ -147,6 +150,31 @@ func (ms *MilestoneDiff) kvStorableLoad(utxoManager *Manager, key []byte, value
 	return nil
 }
 
+func (ms *MilestoneDiff) sortedOutputs() LexicalOrderedOutputs {
+	var sortedOutputs LexicalOrderedOutputs = LexicalOrderedOutputs(ms.Outputs)
+	sort.Sort(sortedOutputs)
+	return sortedOutputs
+}
+
+func (ms *MilestoneDiff) sortedSpents() LexicalOrderedSpents {
+	var sortedSpents LexicalOrderedSpents = LexicalOrderedSpents(ms.Spents)
+	sort.Sort(sortedSpents)
+	return sortedSpents
+}
+
+func (ms *MilestoneDiff) SHA256Sum() ([]byte, error) {
+
+	msDiffHash := sha256.New()
+
+	// compute the sha256 of the milestone diff byte representation
+	if err := binary.Write(msDiffHash, binary.LittleEndian, append(ms.kvStorableKey(), ms.kvStorableValue()...)); err != nil {
```

Comment:
why append first? You can add first the key, then the value to the hash, is probably more memory-efficient